File: Dummy/hawkshaw/library/datalayer/room/AppDatabaseKt.java
package com.hawkshaw.library.datalayer.room;

import B1.a;
import W2.e;
import W2.l;

public final class AppDatabaseKt {
    private static final e db$delegate = new l(a.f174D);

    public static final AppDatabase getDb() {
        return (AppDatabase) db$delegate.getValue();
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/room/AppDatabaseKt.kt
package com.hawkshaw.library.datalayer.room

import androidx.room.Room
import com.hawkshaw.library.App

/**
 * Provides a singleton instance of the Room database
 */
object AppDatabaseKt {
    private const val DATABASE_NAME = "hawkshaw-db"

    val db: AppDatabase by lazy {
        Room.databaseBuilder(
            App.getContext(),
            AppDatabase::class.java,
            DATABASE_NAME
        ).build()
    }
} 